File: app/src/main/java/com/example/nutritionapp/Register/WeightFragment.java
package com.example.nutritionapp.Register;

import static android.content.Context.MODE_PRIVATE;

import android.content.SharedPreferences;
import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.example.nutritionapp.R;
import com.example.nutritionapp.databinding.FragmentWeightBinding;
import com.google.android.material.floatingactionbutton.ExtendedFloatingActionButton;

public class WeightFragment extends Fragment {
    private FragmentWeightBinding binding;
    private ExtendedFloatingActionButton next,back;
    TextView weight;
    ImageView incweight,decweight;
    RegisterMain registerMain;


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentWeightBinding.inflate(inflater, container, false);
        View rootView = binding.getRoot();

        weight=rootView.findViewById(R.id.fragment_weight_currentWeight);
        incweight=rootView.findViewById(R.id.fragment_weight_incrementWeight);
        decweight=rootView.findViewById(R.id.fragment_weight_decrementWeight);
        next= rootView.findViewById(R.id.weight_next);
        back=rootView.findViewById(R.id.weight_back);


        int bundleWeight = 50, bundleHeight = 180, age=18;
        String active = "light";
        //  to get the data
        Bundle bundle = this.getArguments();
        if(bundle!=null) {
            bundleWeight = bundle.getInt("weight", 50);
            bundleHeight = bundle.getInt("height", 180);
            active = bundle.getString("active", "light");

            weight.setText(String.valueOf(bundleWeight));
        }
        if(bundle==null){
            weight.setText("50");
        }
        final int[] weight1 = {50};
        final String[] weight2 = {"50"};
        incweight.setOnClickListener(view -> {
            weight1[0] +=1;
            weight2[0] =String.valueOf(weight1[0]);
            weight.setText(weight2[0]);
        });

        decweight.setOnClickListener(view -> {
            if(weight1[0] >0) {
                weight1[0] -= 1;
                weight2[0] = String.valueOf(weight1[0]);
                weight.setText(weight2[0]);
            }
        });

        registerMain = (RegisterMain) getActivity();

        int finalBundleHeight = bundleHeight;
        String finalActive = active;
        next.setOnClickListener(view -> {
            boolean valid = true;
            if(weight2[0].equals("0"))
            {
                Toast.makeText(getContext(), "Weight cannot be 0", Toast.LENGTH_SHORT).show();
                valid=false;
            }
            if(valid) {
                HeightFragment heightFragment = new HeightFragment();
                sendInfo(heightFragment, finalBundleHeight, finalActive);
                getFragmentManager().beginTransaction()
                        .replace(R.id.register_container, heightFragment)
                        .commit();
                registerMain.myEdit.putString("weight", weight.getText().toString());
                registerMain.myEdit.commit();
            }

        });
        back.setOnClickListener(view -> {
            AgeFragment ageFragment = new AgeFragment();
            sendInfo(ageFragment, finalBundleHeight, finalActive);
            getFragmentManager().beginTransaction()
                    .replace(R.id.register_container, ageFragment)
                    .commit();
        });

        return rootView;
    }

        private void sendInfo(Fragment fragment, int height, String active){

        boolean selectedMale = false;
        boolean selectedFemale = false;
        int sendWeight  = Integer.parseInt(weight.getText().toString());
            /** data  only gets stored in shared preferences if next is clicked
             * other wise take data from editText
             * */

        // get data from SharedPreferences if next/back is pressed
        SharedPreferences sh = requireActivity().getSharedPreferences("MySharedPref", MODE_PRIVATE);
        String name= sh.getString("name", "");
        String email= sh.getString("email", "");
        String mobile= sh.getString("mobile", "");
        String passwd= sh.getString("pass", "");
        String age = sh.getString("age", "");
        String gender = sh.getString("gender", "");

        if(gender.equals("Male"))
                selectedMale = true;
        else
                selectedFemale= true;
        int intAge = Integer.parseInt(age);
        // add data to bundle
        Bundle bundle = new Bundle();
        bundle.putString("name", name);
        bundle.putString("email", email);
        bundle.putString("mobile", mobile);
        bundle.putString("pass", passwd);
        bundle.putBoolean("selectedMale", selectedMale);
        bundle.putBoolean("selectedFemale", selectedFemale);
        bundle.putInt("age", intAge);
        bundle.putInt("weight", sendWeight);
        bundle.putInt("height", height);
        bundle.putString("active", active);
        fragment.setArguments(bundle);
    }
}

File: app/src/main/java/com/example/nutritionapp/Register/WeightFragment.kt
package com.example.nutritionapp.Register

import android.content.Context
import android.content.SharedPreferences
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.fragment.app.Fragment
import com.example.nutritionapp.R
import com.example.nutritionapp.databinding.FragmentWeightBinding
import com.google.android.material.floatingactionbutton.ExtendedFloatingActionButton

class WeightFragment : Fragment() {
    private var binding: FragmentWeightBinding? = null
    lateinit var next: ExtendedFloatingActionButton
    lateinit var back: ExtendedFloatingActionButton
    lateinit var weight: TextView
    lateinit var incweight: ImageView
    lateinit var decweight: ImageView
    var registerMain: RegisterMain? = null
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        binding = FragmentWeightBinding.inflate(inflater, container, false)
        val rootView: View = binding!!.root
        weight = rootView.findViewById(R.id.fragment_weight_currentWeight)
        incweight = rootView.findViewById(R.id.fragment_weight_incrementWeight)
        decweight = rootView.findViewById(R.id.fragment_weight_decrementWeight)
        next = rootView.findViewById(R.id.weight_next)
        back = rootView.findViewById(R.id.weight_back)
        var bundleWeight = 50
        var bundleHeight = 180
        val age = 18
        var active = "light"
        //  to get the data
        val bundle = this.arguments
        if (bundle != null) {
            bundleWeight = bundle.getInt("weight", 50)
            bundleHeight = bundle.getInt("height", 180)
            active = bundle.getString("active", "light")
            weight.setText(bundleWeight.toString())
        }
        if (bundle == null) {
            weight.setText("50")
        }
        val weight1 = intArrayOf(50)
        val weight2 = arrayOf("50")
        incweight.setOnClickListener(View.OnClickListener { view: View? ->
            weight1[0] += 1
            weight2[0] = weight1[0].toString()
            weight.setText(weight2[0])
        })
        decweight.setOnClickListener(View.OnClickListener { view: View? ->
            if (weight1[0] > 0) {
                weight1[0] -= 1
                weight2[0] = weight1[0].toString()
                weight.setText(weight2[0])
            }
        })
        registerMain = activity as RegisterMain?
        val finalBundleHeight = bundleHeight
        val finalActive = active
        next.setOnClickListener(View.OnClickListener { view: View? ->
            var valid = true
            if (weight2[0] == "0") {
                Toast.makeText(context, "Weight cannot be 0", Toast.LENGTH_SHORT).show()
                valid = false
            }
            if (valid) {
                val heightFragment = HeightFragment()
                sendInfo(heightFragment, finalBundleHeight, finalActive)
                requireFragmentManager().beginTransaction()
                        .replace(R.id.register_container, heightFragment)
                        .commit()
                registerMain!!.myEdit!!.putString("weight", weight.getText().toString())
                registerMain!!.myEdit!!.commit()
            }
        })
        back.setOnClickListener(View.OnClickListener { view: View? ->
            val ageFragment = AgeFragment()
            sendInfo(ageFragment, finalBundleHeight, finalActive)
            requireFragmentManager().beginTransaction()
                    .replace(R.id.register_container, ageFragment)
                    .commit()
        })
        return rootView
    }

    private fun sendInfo(fragment: Fragment, height: Int, active: String) {
        var selectedMale = false
        var selectedFemale = false
        val sendWeight = weight!!.text.toString().toInt()

        /** data  only gets stored in shared preferences if next is clicked
         * other wise take data from editText
         */

        // get data from SharedPreferences if next/back is pressed
        val sh = requireActivity().getSharedPreferences("MySharedPref", Context.MODE_PRIVATE)
        val name = sh.getString("name", "")
        val email = sh.getString("email", "")
        val mobile = sh.getString("mobile", "")
        val passwd = sh.getString("pass", "")
        val age = sh.getString("age", "")
        val gender = sh.getString("gender", "")
        if (gender == "Male") selectedMale = true else selectedFemale = true
        val intAge = age!!.toInt()
        // add data to bundle
        val bundle = Bundle()
        bundle.putString("name", name)
        bundle.putString("email", email)
        bundle.putString("mobile", mobile)
        bundle.putString("pass", passwd)
        bundle.putBoolean("selectedMale", selectedMale)
        bundle.putBoolean("selectedFemale", selectedFemale)
        bundle.putInt("age", intAge)
        bundle.putInt("weight", sendWeight)
        bundle.putInt("height", height)
        bundle.putString("active", active)
        fragment.arguments = bundle
    }
}